drivers/meter/ade7758/ade7758.h

```diff
@@ -0,0 +1,515 @@
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES, INC. “AS IS” AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL ANALOG DEVICES, INC. BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef __ADE7758_H__
+#define __ADE7758_H__
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+#include "no_os_util.h"
+#include "no_os_spi.h"
+#include "no_os_gpio.h"
+#include "no_os_irq.h"
+#include "no_os_delay.h"
+#include "no_os_units.h"
+#include "no_os_alloc.h"
+#include "no_os_crc16.h"
+#include "no_os_print_log.h"
+#include <stdlib.h>
+#include <errno.h>
+#include <math.h>
+
+/* SPI commands */
+#define ADE7758_SPI_READ					NO_OS_BIT(7)
+
+/* Version product */
+#define ADE7758_VERSION						0x3F
+/* Use a register that has a non-zero value at reset */
+#define ADE7758_REG_VERSION_PRODUCT     			0x48
+
+/* Miscellaneous Definitions */
+#define IS_7BITS_REG(x) ((x) >= ADE7758_REG_APHCAL && (x) <= ADE7758_REG_CPHCAL)
+#define IS_8BITS_REG(x) (((x) == ADE7758_REG_TEMP) || ((x) >= ADE7758_REG_OPMODE && (x) <= ADE7758_REG_LCYCMODE) \
+        	|| ((x) >= ADE7758_REG_SAGCYC && (x) <= ADE7758_REG_GAIN) || ((x) >= ADE7758_REG_WDIV && (x) <= ADE7758_REG_VADIV) \
+		||((x) >= ADE7758_REG_CHKSUM && (x) <= ADE7758_REG_VERSION))
+#define IS_12BITS_REG(x) (((x) == ADE7758_REG_FREQ) || ((x) >= ADE7758_REG_AVRMSGAIN && (x) <= ADE7758_REG_CVAROS) \
+		|| ((x) == ADE7758_REG_APCFDEN) || ((x) == ADE7758_REG_VARCFDEN))
+#define IS_16BITS_REG(x) (((x) >= ADE7758_REG_AWATTHR && (x) <= ADE7758_REG_CVAHR) \
+		|| ((x) >= ADE7758_REG_ZXTOUT && (x) <= ADE7758_REG_LINECYC) || ((x) == ADE7758_REG_APCFNUM) \
+		|| ((x) == ADE7758_REG_VARCFNUM))
+#define IS_24BITS_REG(x) (((x) >= ADE7758_REG_AIRMS && (x) <= ADE7758_REG_CVRMS) \
+		|| ((x) == ADE7758_REG_WFORM) || ((x) >= ADE7758_REG_MASK && (x) <= ADE7758_REG_RSTATUS))
+
+/* Reset Delay */
+#define ADE7758_RESET_DEL					1
+
+/* ENABLE and DISABLE */
+#define ENABLE							1u
```

Comment:
These are going to name clash with some library pretty soon, delete them and just use 0 and 1 inline wherever you use them.